app/routes.js

```diff
@@ -1,14 +1,47 @@
 import express from 'express';
+import { generateTemplate } from './helpers/templateGenerator';
+import defaultParams from './helpers/defaultParams';
 
 const router = express.Router();
 
 router.get('/', async (req, res) => {
   res.render('index', {});
 });
 
-router.get('/component/:componentName', async (req, res) => {
-  const { componentName } = req.params;
-  res.render(`./components/${componentName}/template`, {});
+router.get('/component/:component', async (req, res) => {
+  const { component } = req.params;
+  const { params } = defaultParams[component];
+
+  generateTemplate({
+    component,
+    componentName: defaultParams[component].componentName,
+    params,
+    type: 'component',
+  });
+
+  res.render('template');
+});
+
+router.post('/component/:component', async (req, res) => {
+  const { component } = req.params;
+  const params = Object.entries(req.body).reduce((acc, [param, value]) => {
+    if (!value) acc[param] = defaultParams[component].params[param];
+    else acc[param] = value;
+    return acc;
+  }, {});
+
+  if (component === 'view-data-bulletlist') {
+    params.data = params.data.split(',');
+  }
```

Comment:
could we move this withing the helper thing above? or the generateTemplate function?